tweetnacl-multiplatform/src/jvmMain/kotlin/net/avianlabs/solana/tweetnacl/TweetNaCl.kt

```
@@ -20,9 +20,9 @@ internal actual fun generateKeyInternal(seed: ByteArray): Ed25519Keypair {
 
 internal actual fun secretBoxInternal(secretKey: ByteArray): TweetNaCl.SecretBox =
   object : TweetNaCl.SecretBox {
-    override fun box(message: ByteArray, nonce: ByteArray): ByteArray =
+    override fun box(message: ByteArray, nonce: ByteArray): ByteArray? =
       TweetNaclFast.SecretBox(secretKey).box(message, nonce)
 
-    override fun open(box: ByteArray, nonce: ByteArray): ByteArray =
+    override fun open(box: ByteArray, nonce: ByteArray): ByteArray? =
       TweetNaclFast.SecretBox(secretKey).open(box, nonce)
```

Doesn't this throw instead of returning null ?

---

Maybe we can runCatching getOrNull here ?

---

Ideally with some logger 👀 :¯\_(ツ)_/¯

---

nope

---

Ah ok. It was throwing nullpointer exception with the result being null 👍

---

sorry the nope was at the throw comment 😂

---

this is a library, I think the functionality should be minimal. I'm doing all those things in the app